radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/CandidateForkVote.java

```diff
@@ -62,61 +62,79 @@
  * permissions under this License.
  */
 
-package com.radixdlt.statecomputer;
+package com.radixdlt.statecomputer.forks;
 
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Streams;
-import com.radixdlt.application.validators.state.ValidatorSystemMetadata;
-import com.radixdlt.atom.SubstateTypeId;
-import com.radixdlt.constraintmachine.REProcessedTxn;
-import com.radixdlt.constraintmachine.RawSubstateBytes;
-import com.radixdlt.constraintmachine.SubstateIndex;
+import com.google.common.hash.HashCode;
+import com.google.common.primitives.Bytes;
 import com.radixdlt.crypto.ECPublicKey;
-import com.radixdlt.crypto.exception.PublicKeyException;
-import com.radixdlt.engine.PostProcessor;
-import com.radixdlt.engine.PostProcessorException;
-import com.radixdlt.store.EngineStore;
+import com.radixdlt.crypto.HashUtils;
+import com.radixdlt.utils.Ints;
+import com.radixdlt.utils.Longs;
+import com.radixdlt.utils.Shorts;
+import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
-import java.util.List;
+import java.util.stream.Stream;
 
-/** Adds validatorsSystemMetadata at epoch boundary to result metadata. */
-public final class ValidatorsSystemMetadataPostProcessor
-    implements PostProcessor<LedgerAndBFTProof> {
-  @Override
-  public LedgerAndBFTProof process(
-      LedgerAndBFTProof metadata,
-      EngineStore<LedgerAndBFTProof> engineStore,
-      List<REProcessedTxn> txns)
-      throws PostProcessorException {
-    if (metadata.getProof().getNextValidatorSet().isPresent()) {
-      return metadata.withValidatorsSystemMetadata(
-          getValidatorsSystemMetadata(engineStore, metadata));
-    } else {
-      return metadata;
-    }
+public record CandidateForkVote(HashCode payload) {
+  public static final HashCode FORK_VOTE_NONCE =
+      HashUtils.sha256("olympia".getBytes(StandardCharsets.US_ASCII));
```

Comment:
why not UTF-8?

Reply:
it doesn't really matter, nonce is just some constant seq of bytes